bin/output.ml

```diff
@@ -55,22 +55,47 @@ let pp_cram ppf (t:Mdx.Cram.t) =
     (pp_list pp_line) t.command
     (pp_list pp_output) t.output pp_exit
 
-let pp_block ppf (b:Mdx.Block.t) =
-  let lang, pp_code, attrs = match b.value with
-    | Toplevel t -> Some "ocaml", (fun ppf -> pp_list pp_toplevel ppf t), [
+let try_converting_to_reason (b:Mdx.Block.t) : Mdx.Block.t =
+  let possibly_reason_contents = Mdx.Reason.FromOCamlToReason.transform b.contents in
+  match possibly_reason_contents with
+  | Some contents -> { b with value = Reason; contents = contents; }
+  | None -> b
+
+let try_converting_to_ocaml (b:Mdx.Block.t) : Mdx.Block.t =
+  let possibly_ocaml_contents = Mdx.Reason.FromReasonToOCaml.transform b.contents in
+  match possibly_ocaml_contents with
+  | Some contents -> { b with value = OCaml; contents = contents; }
+  | None -> b
+
+let pp_block ~print_form ppf (b:Mdx.Block.t) =
+  let lang, pp_code, attrs = match b.value, print_form with
+    | Toplevel t, _ -> Some "ocaml", (fun ppf -> pp_list pp_toplevel ppf t), [
         ("class"             , "command-line");
         ("data-prompt"       , "#");
         ("data-filter-output", ">");
       ]
-    | OCaml  -> Some "ocaml", pp_contents b, []
-    | Cram t -> Some "bash" , (fun ppf -> pp_list pp_cram ppf t.tests), [
+    | OCaml, `OCaml -> Some "ocaml", pp_contents b, []
+    | OCaml, `Reason -> (
+      let new_block = try_converting_to_reason b in
+      match new_block.value with
+        | Reason -> Some "reason", pp_contents new_block, []
+        | _ -> Some "ocaml", pp_contents b, [] (* transform to reason failed, so let's fallback to ocaml *)
```

Comment:
Under what circumstances would a transformation from OCaml to Reason syntax fail?  Dropping back silently seems like a bad idea as it would leave some blocks in different syntaxes.

Reply:
I did it, because in https://github.com/realworldocaml/book there are some examples which doesn't have a valid syntax (mostly intentionally, I guess).

For example
```ocaml
val remember : '_a -> '_a = <fun>
```
in Imperative Programming chapter.

It made sense in context of transforming the book, but I agree that for mdx as a standalone tool it's confusing. Do you want me to change it to fail fast?

Reply:
> Do you want me to change it to fail fast?

We do need some alternate syntax for failing syntax fragments indeed.  I'm not sure what that is yet... some sort of alternates mode in mdx seems necessary to pass this through while selecting the right language.

Reply:
What about adding label to code blocks?
For example
```
```ocaml unparsable
```

Reply:
these unparsable bits are normally wrapped with a `skip` keyword to tell mdx to ignore them

Reply:
I do think we need some alternates mechanism for this. I'm not keen on a 95% solution that'll leave some OCaml syntax lingering around to confuse a ReasonML-only reader...